pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -619,7 +660,7 @@ func TestReconcile(t *testing.T) {
 			)
 
 			k8sclient := builder.Build()
-			recorder := record.NewBroadcaster().NewRecorder(k8sclient.Scheme(), corev1.EventSource{Component: "admission-checks-controller"})
+			recorder := &utiltesting.EventTestRecorder{}
```

Comment:
For testing, can we use the record.FakeRecorder: https://github.com/kubernetes/client-go/blob/84a6fe7e4032ae1b8bc03b5208e771c5f7103549/tools/record/fake.go#L80

I see it used in k8s here: https://github.com/kubernetes/kubernetes/blob/0c645922edcc06adff43c70c02fb56751364bbb5/pkg/kubelet/active_deadline_test.go#L52

Reply:
it only keeps a formatted string, we cannot check related object and so on and ignore parts of the event with cmpopt.

Reply:
Yes, I just want to avoid adding our custom code, unless there is a good reason. The formatted string combines the information from all the fields: https://github.com/kubernetes/client-go/blob/84a6fe7e4032ae1b8bc03b5208e771c5f7103549/tools/record/fake.go#L57C27-L57C36 

Are there some test cases we would not be able to cover by asserting on the string?

Reply:
How annoying. I prefer we check the individual fields.

Reply:
> Are there some test cases we would not be able to cover by asserting on the string?

- Not in this PR, but the scheduler creates events that contain the time it took to admit, which is not easy to hardcode.
- Also the formatted string will only contain the GVK of the related object not its name.